p3/src/block_engine/block_engine.rs

```diff
@@ -0,0 +1,694 @@
+                    Err(TrySendError::Full(_)) => {
+                        relayer_metrics.num_try_send_channel_full += 1;
+                        warn!("heartbeat channel is full for: {:?}", pubkey);
+                    }
+                }
+                None
+            })
+            .collect();
+
+        failed_pubkey_updates
+    }
+
+    /// Returns pubkeys of subscribers that failed to send
+    fn forward_packets(
+        maybe_packet_batch: Result<PacketBatch, RecvError>,
+        packet_subscriptions: &PacketSubscriptions,
+        bundle_subscriptions: &BundleSubscriptions,
+        relayer_metrics: &mut BlockEngineMetrics,
+    ) -> BlockEngineResult<Vec<Pubkey>> {
+        let packet_batch = maybe_packet_batch?;
+        let packets: Vec<_> = packet_batch
+            .iter()
+            .filter_map(packet_to_proto_packet)
+            .collect();
+        let batch = ProtoPacketBatch { packets };
+        let num_packets = batch.packets.len() as u64;
+        let mut failed_forwards = Vec::new();
+
+        fn broadcast<T: Clone>(
+            entries: Vec<(&Pubkey, &TokioSender<Result<T, Status>>)>,
+            make_resp: impl Fn(&Pubkey) -> T,
+            metrics: &mut BlockEngineMetrics,
+            count: u64,
+        ) -> Vec<Pubkey> {
+            let mut failed = Vec::new();
+            for (pubkey, sender) in entries {
+                let resp = make_resp(pubkey);
+                match sender.try_send(Ok(resp)) {
+                    Ok(_) => metrics.increment_packets_forwarded(pubkey, count),
+                    Err(TrySendError::Full(_)) => metrics.increment_packets_dropped(pubkey, count),
+                    Err(TrySendError::Closed(_)) => failed.push(*pubkey),
+                }
+            }
+            failed
+        }
+
+        // send packets to packet_subscriptions
+        let l_senders = packet_subscriptions.read().unwrap();
+        let packet_senders = l_senders.iter().collect::<Vec<_>>();
+        failed_forwards.extend(broadcast(
+            packet_senders,
+            |_pubkey| SubscribePacketsResponse {
+                header: Some(Header {
+                    ts: Some(Timestamp::from(SystemTime::now())),
+                }),
+                batch: Some(batch.clone()),
+            },
+            relayer_metrics,
+            num_packets,
+        ));
+
+        // build and send bundles to bundle_subscriptions
+        let bundle = packet_batch_to_bundle(&packet_batch, SystemTime::now(), String::default())
+            .unwrap_or_default();
+        let l_senders = bundle_subscriptions.read().unwrap();
+        let bundle_senders = l_senders.iter().collect::<Vec<_>>();
+        failed_forwards.extend(broadcast(
+            bundle_senders,
+            |_| SubscribeBundlesResponse {
+                bundles: vec![bundle.clone()],
+            },
+            relayer_metrics,
+            num_packets,
+        ));
```

Comment:
We you sending ever p3 packet as both a packet and a bundle? Please only send packets with `meta().is_mev()` as bundles. the mev flagged packets must only be sent as bundles and not  as packets